packages/graphql/tests/integration/issues/2670.int.test.ts

```diff
@@ -0,0 +1,594 @@
+                {
+                    title: movieTitle2,
+                },
+                {
+                    title: movieTitle3,
+                },
+            ]),
+        });
+    });
+
+    test("should find where moviesAggregate node property AVERAGE", async () => {
+        const query = `
+            {
+                ${movieType.plural}(where: { genresConnection: { node: { moviesAggregate: { node: { title_AVERAGE_EQUAL: ${genre2AverageTitleLength} } } } } }) {
+                    title
+                }
+            }
+        `;
+
+        const result = await graphql({
+            schema: await neoSchema.getSchema(),
+            source: query,
+            contextValue: neo4j.getContextValues(),
+        });
+
+        expect(result.errors).toBeFalsy();
+        expect(result.data).toEqual({
+            [movieType.plural]: expect.toIncludeSameMembers([
+                {
+                    title: movieTitle4,
+                },
+                {
+                    title: movieTitle2,
+                },
+            ]),
+        });
+    });
+
+    test("should find where moviesAggregate edge property MAX_LT", async () => {
+        const query = `
+            {
+                ${movieType.plural}(where: { genresConnection: { node: { moviesAggregate: { edge: { intValue_MAX_LT: ${intValue3} } } } } }) {
+                    title
+                }
+            }
+        `;
+
+        const result = await graphql({
+            schema: await neoSchema.getSchema(),
+            source: query,
+            contextValue: neo4j.getContextValues(),
+        });
+
+        expect(result.errors).toBeFalsy();
+        expect(result.data).toEqual({
+            [movieType.plural]: expect.toIncludeSameMembers([
+                {
+                    title: movieTitle4,
+                },
+                {
+                    title: movieTitle2,
+                },
+            ]),
+        });
+    });
+
+    test("should find where moviesAggregate edge property MIN_EQUAL", async () => {
+        const query = `
+            {
+                ${movieType.plural}(where: { genresConnection: { node: { moviesAggregate: { edge: { intValue_MIN_EQUAL: ${intValue1} } } } } }) {
+                    title
+                }
+            }
+        `;
+
+        const result = await graphql({
+            schema: await neoSchema.getSchema(),
+            source: query,
+            contextValue: neo4j.getContextValues(),
+        });
+
+        expect(result.errors).toBeFalsy();
+        expect(result.data).toEqual({
+            [movieType.plural]: expect.toIncludeSameMembers([
+                {
+                    title: movieTitle1,
+                },
+                {
+                    title: movieTitle2,
+                },
+                {
+                    title: movieTitle3,
+                },
+            ]),
+        });
+    });
+
+    test("should find where genresConnection_SOME", async () => {
```

Comment:
A bit more context here maybe?
```suggestion
    test("should find where genresConnection_SOME and nested count", async () => {
```